doc/crypto/api/ops/key-agreement.rst

```diff
@@ -332,6 +338,264 @@ Combining key agreement and key derivation
+
+    .. return:: uint32_t
+        Number of *ops* that the operation has taken so far.
+
+    After the interruptible operation has completed, the returned value is the number of *ops* required for the entire operation.
+    The value is reset to zero by a call to either `psa_key_agreement_iop_setup()` or `psa_key_agreement_iop_abort()`.
+
+    This function can be used to tune the value passed to `psa_iop_set_max_ops()`.
+
+    The value is undefined if the operation object has not been initialized.
+
+.. function:: psa_key_agreement_iop_setup
+
+    .. summary::
+        Start an interruptible operation to perform a key agreement.
+
+    .. param:: psa_key_agreement_iop_t * operation
+        The interruptible key agreement operation to set up.
+        It must have been initialized as per the documentation for `psa_key_agreement_iop_t`, and be inactive.
+    .. param:: psa_key_id_t private_key
+        Identifier of the private key to use.
+        It must permit the usage `PSA_KEY_USAGE_DERIVE`.
+    .. param:: const uint8_t * peer_key
+        Public key of the peer.
+        The peer key data is parsed with the type :code:`PSA_KEY_TYPE_PUBLIC_KEY_OF_KEY_PAIR(type)` where ``type`` is the type of ``private_key``, and with the same bit-size as ``private_key``.
+        The peer key must be in the format that `psa_import_key()` accepts for this public key type.
+        These formats are described with the public key type in :secref:`key-types`.
+    .. param:: size_t peer_key_length
+        Size of ``peer_key`` in bytes.
+    .. param:: psa_algorithm_t alg
+        The standalone key agreement algorithm to compute: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_STANDALONE_KEY_AGREEMENT(alg)` is true.
+    .. param:: const psa_key_attributes_t * attributes
+        The attributes for the new key.
+        This function uses the attributes as follows:
+
+        *   The key type must be one of `PSA_KEY_TYPE_DERIVE`, `PSA_KEY_TYPE_RAW_DATA`, `PSA_KEY_TYPE_HMAC`, or `PSA_KEY_TYPE_PASSWORD`.
+
+            Implementations must support the `PSA_KEY_TYPE_DERIVE` and `PSA_KEY_TYPE_RAW_DATA` key types.
+
+        *   The size of the returned key is always the bit-size of the shared secret, rounded up to a whole number of bytes.
+            The key size in ``attributes`` can be zero; if it is nonzero, it must be equal to the output size of the key agreement, in bits.
+
+            The output size, in bits, of the key agreement is :code:`8 * PSA_RAW_KEY_AGREEMENT_OUTPUT_SIZE(type, bits)`, where ``type`` and ``bits`` are the type and bit-size of ``private_key``.
+
+        *   The key permitted-algorithm policy is required for keys that will be used for a cryptographic operation, see :secref:`permitted-algorithms`.
+        *   The key usage flags define what operations are permitted with the key, see :secref:`key-usage-flags`.
+        *   The key lifetime and identifier are required for a persistent key.
```

Comment:
The lifetime and identifier are also required for a volatile key, they just use different sets of values.

Reply:
That text is taken from _all_ the other key creation functions, and was first used when we clarified the use of attributes 4 years ago. (Private link to original source commit: https://github.com/ARM-software/psa-crypto-api/commit/5d12b9abdcb1cc183a0ba8113771a034ecc6fd64).

Perhaps it would be clearer to state that the lifetime and identifier have to be set explicitly for a persistent key, as the default values will create a volatile key?

Reply:
Proposal for improvement raised as #205